game-core/src/test/java/games/strategy/engine/message/RemoteActionCodeTest.java

```diff
@@ -0,0 +1,49 @@
+package games.strategy.engine.message;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.lang.reflect.Method;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.aggregator.AggregateWith;
+import org.junit.jupiter.params.aggregator.ArgumentsAccessor;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregationException;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregator;
+import org.junit.jupiter.params.provider.CsvFileSource;
+
+public class RemoteActionCodeTest {
+
+  @ParameterizedTest
+  @CsvFileSource(resources = "/required-op-codes.txt")
+  void verifyCorrectOpCode(
+      final int opCode, @AggregateWith(MethodAggregator.class) final Method method) {
+    final var remoteActionCode = method.getAnnotation(RemoteActionCode.class);
+
+    assertThat("No annotation present for " + method, remoteActionCode, is(notNullValue()));
+
+    assertThat("Invalid value for " + method, remoteActionCode.value(), is(opCode));
+  }
+
+  static class MethodAggregator implements ArgumentsAggregator {
+    @Override
+    public Method aggregateArguments(
+        final ArgumentsAccessor arguments, final ParameterContext context)
+        throws ArgumentsAggregationException {
+      final int offset = context.getIndex();
```

Comment:
Can this offset variable be clarified a bit more, what are we offsetting from? It might be possible to rename the variable, or some inline commentary could help explain it.

Reply:
It's a bit hard to explain. Basically `ArgumentsAccessor` has access to the Arguments that _could_ be passed to a specific method but the opcode (the first column in our csv file) has nothing to do with the actual method.
We could hardcode this offset to be 1 (so we ignore the first column of the csv) but because we already "consume" this parameter by explicitly declaring it in the test method I thought it would make sense to just ignore as many columns as there are parameters before Method.

Might be confusing, but I hope this helps a little bit:
```java
void verifyCorrectOpCode(/* parameter context index 0*/final int opCode, @AggregateWith(MethodAggregator.class) /* parameter context index 1*/final Method method) { /* Test code */}

// What argumentsAccessor holds, based off csv content:
arguments.getInt(0); // Op code
arguments.get(1, Class.class); // Subinterface of IRemote of IChannelSubscriber
arguments.getString(2); // method name

arguments.get(3, Class.class); // start method parameter types
.
.
.
arguments.get(arguments.size() - 1, Class.class); // end method parameter types
```

Reply:
Can we move this clarification into the code somehow? It's always very sad when you have to look at the date of a merge to find a PR to then read comments to understand the code you're looking at.